### PR TITLE
Ficus rg fixes

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Merge  some fix to  ficus-rg. There are:
- Add email field as compulsory in create/edit user form
- Fix save textbook - fix texbook in studio. Saving a large number of chapters,
- Fix problem with delete/register user in forum with equal data - Change the user login in the forum to a unique value. Since there is no IP to delete the user. Temporary fix. Because the login of the user in the forum is clickable.
- fix problem with empty option icon menu - Now: if the course is completed then you can not unsubscribe from the course
- set setting: MAX_ASSET_UPLOAD_FILE_SIZE_IN_MB = 50
- fix problem with enable instructor tabs for student in preview
- Add unique validation for email
- short_description max_length=150
- Add reload page after the course was reruned
- Add catalog_visibility for course discovery

**Youtrack:** https://youtrack.raccoongang.com/issue/CB-46

**Trello**  https://trello.com/b/kmuKODcC/common-bugs

### CODE REVIEW

**Comment by AndreyLykhoman** [comment]
start unit tests

**Comment by hetmantsev** [comment]
start unit tests

**Comment by AndreyLykhoman** [comment]
start unit tests

**Comment by AndreyLykhoman** [comment]
start unit tests

File: cms/djangoapps/contentstore/api/tests/test_view.py [hunk 1]
```diff
@@ -89,10 +89,15 @@ def test_check_rerun_courses(self, options):
...
+        with patch(
+                'cms.djangoapps.contentstore.api.views.has_studio_read_access',
+                return_value=options['has_studio_read_access']
+        ):
+            with patch(
+                    'cms.djangoapps.contentstore.api.views.CourseRerunState.objects',
+                       course_rerun_mock
+            ) :
                 resp = self.client.post('/api/courses/v0/check_rerun_courses/', {'courses': options['courses_ids']})
                 self.assertEqual(resp.content, '{}'.format(options['is_reload']))
```
Comment (idegtiarov) [inline cms/djangoapps/contentstore/api/tests/test_view.py]: trailing empty line is required.

File: cms/djangoapps/contentstore/api/tests/test_view.py [hunk 2]
```diff
@@ -89,10 +89,15 @@ def test_check_rerun_courses(self, options):
...
             mock.course_key = course_id
             find_all.append(mock)
-        side_mock_auth = Mock()
-        side_mock_auth.has_studio_read_access_value = options['has_studio_read_access']
-        with patch('student.auth.has_studio_read_access', side_mock_auth):
-            with patch('course_action_state.models.CourseRerunState.objects.find_all',
-                       Mock(side_effect=lambda exclude_args,should_display: find_all )) as course_rerun_mock:
+        course_rerun_mock = Mock()
+        course_rerun_mock.find_all.return_value = find_all
+        with patch(
```
Comment (idegtiarov) [inline cms/djangoapps/contentstore/api/tests/test_view.py]: It is rather a complex construction context manager onside context manager, I would use at least one patch as the decorator of the test method, but it is up to you.

**Pull request merged**